HousingSearchListener.Tests/V1/E2ETests/Stories/AddOrUpdateContractOnAssetTests.cs

```diff
@@ -79,5 +78,21 @@ public void ContractAddedToAsset(string eventType)
                     _ContractsApiFixture.ResponseObject, _esFixture.ElasticSearchClient))
                 .BDDfy();
         }
+
+        [Theory]
+        [InlineData(EventTypes.ContractCreatedEvent)]
+        [InlineData(EventTypes.ContractUpdatedEvent)]
+        public void ContractNotAddedToAssetWhenNoUnapprovedContractsArePresent(string eventType)
+        {
+            var contractId = Guid.NewGuid();
+            var assetId = Guid.NewGuid();
+            this.Given(g => _ContractsApiFixture.GivenApprovedContractsAreReturned(contractId, assetId))
+                .And(g => _AssetApiFixture.GivenTheAssetExists(assetId))
+                .And(g => _esFixture.GivenAnAssetIsIndexed(assetId.ToString()))
+                .When(w => _steps.WhenTheFunctionIsTriggered(contractId, eventType, assetId.ToString()))
+                .Then(t => _steps.ThenTheAssetInTheIndexIsUpdatedAndHasNoContracts(_AssetApiFixture.ResponseObject,
+                    _esFixture.ElasticSearchClient))
```

Comment:
1. `Given Approved Contracts Are Returned`
2. `Then The Asset In The Index Is Updated And Has No Contracts`

Is this what is expected?